packages/emitter/src/index.ts

```diff
@@ -16,6 +16,26 @@ import {
 
 type Quote = '"' | "'";
 
+interface EmitOptions {
+  /**
+   * The preferred quote character to use when `emit` encounters a comparison value that needs to be escaped by wrapping
+   * in quotes. Either `"` (the ASCII double quote character) or `'` (the ASCII single quote character). Defaults to `"`
+   * (the ASCII double quote character).
+   */
+  preferredQuote?: Quote;
+  /**
+   * If `true`, `emit` will override the `preferredQuote` setting on a comparison value-by-comparison value basis if
+   * doing so would shorten the emitted RSQL. If `false`, `emit` will use the `preferredQuote` as the quote character
+   * for all comparison values encountered. Defaults to `true`.
+   */
+  optimizeQuotes?: boolean;
+}
+
+const DEFAULT_EMIT_OPTIONS: Required<EmitOptions> = {
+  preferredQuote: '"',
+  optimizeQuotes: true,
```

Comment:
@piotr-oles I went back and forth on what default to use for `optimizeQuotes`. To make the new quoting behaviors 100% opt in, the correct defaults would be `preferredQuote: '"'` + `optimizeQuotes: false`. I decided to go with `optimizeQuotes: true` instead b/c a) it makes sense to me to default to the shortest possible RSQL for the given input, and b) I'm guessing most callers of `emit` don't actually care about double vs single quotes and wouldn't consider a switch from double quotes to single quotes for the same input to be a breaking change.

But I can see the case for being conservative and defaulting to `optimizeQuotes: false` instead. Happy to discuss/tweak further. Thanks!

Reply:
I think that's fine - the emitter API doesn't define which quote it will output :)